### PR TITLE
Add pre-commit

### PR DESCRIPTION
**Pull request by Freed-Wu**
Modified from rejected https://github.com/pre-commit/pre-commit/pull/2506

- [x] This compdef is not already available in zsh.
- [x] This compdef is not already available in their original project.
- [x] I am the original author, or I have authorization to submit this work.
- [x] This is a finished work.
- [x] It has a header containing authors, status and origin of the script.
- [x] It has a license header or I accept that it will be licensed under the terms of the Zsh license.

### CODE REVIEW

File: src/_pre-commit [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        init-templatedir) _arguments -C $_pre_commit_init-templatedir_options ;;
+        install) _arguments -C $_pre_commit_install_options ;;
+        install-hooks) _arguments -C $_pre_commit_install-hooks_options ;;
+        migrate-config) _arguments -C $_pre_commit_migrate-config_options ;;
+        run) _arguments -C $_pre_commit_run_options ;;
+        sample-config) _arguments -C $_pre_commit_sample-config_options ;;
+        try-repo) _arguments -C $_pre_commit_try-repo_options ;;
+        uninstall) _arguments -C $_pre_commit_uninstall_options ;;
+        validate-config) _arguments -C $_pre_commit_validate-config_options ;;
+        validate-manifest) _arguments -C $_pre_commit_validate-manifest_options ;;
```
Comment (syohex) [inline src/_pre-commit:200]:
```suggestion
        install-hooks) _arguments -C $_pre_commit_install_hooks_options ;;
        migrate-config) _arguments -C $_pre_commit_migrate_config_options ;;
        run) _arguments -C $_pre_commit_run_options ;;
        sample-config) _arguments -C $_pre_commit_sample_config_options ;;
        try-repo) _arguments -C $_pre_commit_try_repo_options ;;
        uninstall) _arguments -C $_pre_commit_uninstall_options ;;
        validate-config) _arguments -C $_pre_commit_validate_config_options ;;
        validate-manifest) _arguments -C $_pre_commit_validate_manifest_options ;;
```

File: src/_pre-commit [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    "init-templatedir:"
+    "install:"
+    "install-hooks:"
+    "migrate-config:"
+    "run:"
+    "sample-config:"
+    "try-repo:"
+    "uninstall:"
+    "validate-config:"
+    "validate-manifest:"
```
Comment (syohex) [inline src/_pre-commit:32]: Could you add the description of subcommands ?

Comment (Freed-Wu) [reply src/_pre-commit]: Fixed.

**Comment by S1ngularity96** [comment]
Thanks for contributing. I needed this.

File: src/_pre-commit [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    "uninstall:Uninstall the pre-commit script"
+    "validate-config:Validate .pre-commit-config.yaml files"
+    "validate-manifest:Validate .pre-commit-hooks.yaml files"
+  )
+  _describe 'pre-commit commands' _commands
+}
+
+_pre_commit_options=(
+  "(- : *)"{-h,--help}"[show this help message and exit]"
+  "(- : *)--print-completion[print shell completion script]:print_completion:(bash zsh tcsh)"
```
Comment (syohex) [inline src/_pre-commit:39]: It seems `pre-commit` doesn't have `--print-completion` option.

Comment (Freed-Wu) [reply src/_pre-commit]: Fixed.

**Pull request merged**